demosplan/DemosPlanCoreBundle/Entity/Permission/AccessControlPermission.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of the package demosplan.
+ *
+ * (c) 2010-present DEMOS plan GmbH, for more information see the license file.
+ *
+ * All rights reserved
+ */
+
+namespace demosplan\DemosPlanCoreBundle\Entity\Permission;
+
+use DateTime;
+use DemosEurope\DemosplanAddon\Contracts\Entities\AccessControlPermissionInterface;
+use DemosEurope\DemosplanAddon\Contracts\Entities\CustomerInterface;
+use DemosEurope\DemosplanAddon\Contracts\Entities\OrgaInterface;
+use DemosEurope\DemosplanAddon\Contracts\Entities\RoleInterface;
+use DemosEurope\DemosplanAddon\Contracts\Entities\UuidEntityInterface;
+use demosplan\DemosPlanCoreBundle\Entity\CoreEntity;
+use Doctrine\ORM\Mapping as ORM;
+use Gedmo\Mapping\Annotation as Gedmo;
+
+/**
+ * Stores the information about the current phase of a procedure.
+ * Currently there a two phases related to a procedure, therefore this Entity is related to the procedure twice.
+ *
+ * @ORM\Table(name="access_control_permissions")
+ *
+ * @ORM\Entity(repositoryClass="demosplan\DemosPlanCoreBundle\Repository\AccessControlPermissionRepository")
+ */
+class AccessControlPermission extends CoreEntity implements UuidEntityInterface, AccessControlPermissionInterface
+{
+    /**
+     * @ORM\Column(type="string", length=36, options={"fixed":true})
+     *
+     * @ORM\Id
+     *
+     * @ORM\GeneratedValue(strategy="CUSTOM")
+     *
+     * @ORM\CustomIdGenerator(class="\demosplan\DemosPlanCoreBundle\Doctrine\Generator\UuidV4Generator")
+     */
+    protected string $id;
+
+    /**
+     * @ORM\Column(name="permission", type="string", nullable=false)
+     */
+    protected string $permission = '';
+
+    /**
+     * @Gedmo\Timestampable(on="create")
```

Comment:
Check new version, done

Reply:
minor: Im not sure if we want to define a DB-sited default value.
In case of inserting new rows by bypassing doctrine, like an direct SQL, the current timestamp would also be set.
We do not define this as best practice or something afaik, so this is more a question than a request.
```
    /**
     * @Gedmo\Timestampable(on="create")
     *
     * @ORM\Column(type="datetime", nullable=false,  options={"default":"CURRENT_TIMESTAMP"})
     */
    private DateTime $creationDate;
```

Reply:
additionally: I think the @var ist not needed in case of typed property. (@MoritzMandler)